numpy/_core/_dtype.py

```diff
@@ -126,7 +126,7 @@ def _scalar_str(dtype, short):
         else:
             return "'%sU%d'" % (byteorder, dtype.itemsize / 4)
 
-    elif dtype.type == str:
+    elif dtype.type is str:
```

Comment:
Hmm, this might change behavior. @seberg Thoughts?

Reply:
If this was `dtype ==` then yes, as is.  This is fine, although it is a bit funny to not use it above as well.